app/models/compute_resources/foreman/model/gce.rb

```diff
@@ -83,7 +83,7 @@ def new_vm(args = {})
         args.except!(:network)
       end
 
-      if to_bool(args.delete(:associate_external_ip))
+      if to_bool(args[:associate_external_ip])
```

Comment:
Why we are keeping the flag in arguments now? Do we need it later?

Reply:
This `new_vm` method is being called from `create_vm` as well as [.erb views](https://github.com/theforeman/foreman/blob/6c1f956348bcc793adf767a03bc20cef884dc752/app/views/hosts/_compute.html.erb#L1) 

Though this flag is not needed at the time of instance creation i.e from `create_vm` but it is required for views to show current value of this flag. Previously, when tried to remove this flag from `args` then it was failing to show previously selected value of it in case of any failure.

Reply:
Ah, right, it is passed all the way from the params, than good catch! :+1: